### ISSUE TITLE
enh: filter tags

### ISSUE THREAD

**Issue by y9c** [brentp/pbr#7]
Will it be useful to add filter for tags, such as "[NH]>1" to filter reads of multiple hits? Will the same grammar as `--input-fmt-option` of samtools help?

Thank you.

**Comment by brentp** [comment]
I like this idea. It would be lua syntax, like `read.NH > 1` or more likely `tag.NH > 1`.

**Comment by brentp** [comment]
is this or any other issue blocking for you? I probably can't implement everything right now, but would like to get the top priorities started.

**Comment by y9c** [comment]
Thank you @brentp, filtering by tags should be one of the prior features I wish to have. 

Another two major things are:

- The performance, especially for locus with extremely high sequence depth (eg, rRNA sites). I notice that the pileup would become extremely slow. If I remember clearly, the rust htslib toke more than one hour to pileup sites on a rRNA gene (~10kb). I am not sure if this problem can be solved without replacing the htslib? 
- filter mutation. It is not hard to filter the number of mutations based on the "ref_base" column using some "if-else" statements. But it would be great if pbr can support this one out of box.

**Comment by y9c** [comment]
Hi @brentp, thank you for working on this. I would like to known if filtering by tags is supported now?

**Comment by brentp** [comment]
No, for now, I have put this project on hold. I think there is too much overhead in calling a lua expression for each base. Maybe it's possible with using some kind of macro based expressions within rust, but I don't have plans to implement further. 
I found that using pysam with custom code is faster than `pbr`.

**Comment by brentp** [comment]
this is implemented as, e.g. `read:tag('NH`) > 1`
